src/pages/ProfilePage.js

```diff
@@ -0,0 +1,237 @@
+import MenuItem from '../components/MenuItem';
+import AttachmentModal from '../components/AttachmentModal';
+import PressableWithoutFocus from '../components/PressableWithoutFocus';
+import * as Report from '../libs/actions/Report';
+import OfflineWithFeedback from '../components/OfflineWithFeedback';
+import AutoUpdateTime from '../components/AutoUpdateTime';
+import * as UserUtils from '../libs/UserUtils';
+import * as PersonalDetails from '../libs/actions/PersonalDetails';
+import FullScreenLoadingIndicator from '../components/FullscreenLoadingIndicator';
+import BlockingView from '../components/BlockingViews/BlockingView';
+import * as Illustrations from '../components/Icon/Illustrations';
+import variables from '../styles/variables';
+import ROUTES from '../ROUTES';
+
+const matchType = PropTypes.shape({
+    params: PropTypes.shape({
+        /** accountID passed via route /a/:accountID */
+        accountID: PropTypes.string,
+
+        /** report ID passed */
+        reportID: PropTypes.string,
+    }),
+});
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** The personal details of all users */
+    personalDetails: personalDetailsPropType,
+
+    /** Route params */
+    route: matchType.isRequired,
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Date login was validated, used to show info indicator status */
+        validatedDate: PropTypes.string,
+
+        /** Field-specific server side errors keyed by microtime */
+        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    // When opening someone else's profile (via deep link) before login, this is empty
+    personalDetails: {},
+    loginList: {},
+};
+
+/**
+ * Gets the phone number to display for SMS logins
+ *
+ * @param {Object} details
+ * @param {String} details.login
+ * @param {String} details.displayName
+ * @returns {String}
+ */
+const getPhoneNumber = (details) => {
+    // If the user hasn't set a displayName, it is set to their phone number, so use that
+    const displayName = lodashGet(details, 'displayName', '');
+    const parsedPhoneNumber = parsePhoneNumber(displayName);
+    if (parsedPhoneNumber.possible) {
+        return parsedPhoneNumber.number.e164;
+    }
+
+    // If the user has set a displayName, get the phone number from the SMS login
+    return details.login ? Str.removeSMSDomain(details.login) : '';
+};
+
+function ProfilePage(props) {
+    const accountID = lodashGet(props.route.params, 'accountID', 0);
+
+    useEffect(() => {
+        PersonalDetails.openPublicProfilePage(accountID);
+    }, [accountID]);
+
+    const details = lodashGet(props.personalDetails, accountID, {});
+    const displayName = details.displayName ? details.displayName : props.translate('common.hidden');
+    const avatar = lodashGet(details, 'avatar', UserUtils.getDefaultAvatar());
```

Comment:
possibly fine to leave this for now, buttttt I think `getDefaultAvatar` is expecting a login, though we're updating that to be `accountID` in the other migration PR 🤔

Reply:
It allows sending an empty string and then it'll just come up with some default, so I feel like that works here.